addons/mail/static/src/core/common/im_status_service.js

```diff
@@ -43,15 +41,31 @@ export const imStatusService = {
             }
         };
         bus_service.addEventListener("connect", () => updateBusPresence(), { once: true });
-        bus_service.subscribe("bus.bus/im_status_updated", async ({ partner_id, im_status }) => {
-            if (session.is_public || !partner_id || partner_id !== user.partnerId) {
-                return;
+        bus_service.subscribe(
+            "bus.bus/im_status_updated",
+            async ({ im_status, partner_id, guest_id }) => {
+                console.log("getting bus message bus.bus/im_status_updated");
+                const store = env.services["mail.store"];
+                if (!store) {
+                    return;
+                }
+                const persona = store.Persona.get({
+                    type: partner_id ? "partner" : "guest",
+                    id: partner_id ?? guest_id,
+                });
+                if (!persona) {
+                    return; // Do not store unknown persona's status
+                }
+                persona.im_status = im_status;
+                if (persona.notEq(store.self)) {
+                    return;
+                }
+                const isOnline = presence.getInactivityPeriod() < AWAY_DELAY;
+                if ((im_status === "away" && isOnline) || im_status === "offline") {
+                    this.updateBusPresence();
```

Comment:
`this.updateBusPresence` was just a trick not to duplicate code/to provide access to override. It should be removed from `this` and directly called.